app/services/complete_moab_handler.rb

```diff
@@ -186,6 +186,7 @@ def moab_validator
     @moab_validator ||= MoabValidator.new(druid: druid, storage_location: storage_location, results: results)
   end
 
+  # only call this method if a CompleteMoab does not already exist for the druid
```

Comment:
i don't think this will be true, necessarily.  otherwise, we wouldn't have an easy path for getting a second moab for a druid into the catalog (hence the switch to `PreservedObject.find_or_create_by!` instead of always assuming `.create!` is ok).

unfortunately there wasn't a good test in place to break for this already -- will ticket that as specific follow on work to #1535 (and have a failing WIP example test i can push up on a branch).

Reply:
It is currently empirically true, based on the code as I read it.